### PR TITLE
CAMS-282 restrict data verification

### PR DESCRIPTION
**Pull request by btposey**
# Purpose

Restrict data verification to users assigned the data verification role.

# Major Changes

* Added guard to header to dynamically show/hide the data verification navigation.
* Added logic to API to throw unauthorized errors if an unauthorized user attempts to reject/approve orders.
* Appended current user in context to audit records.
* Show user in audit detail (existing logic)

# Testing/Validation

Unit tests above coverage threshold.

# Definition of Done:

- [ ] Acceptance criteria met - The system meets all acceptance criteria
- [ ] Feature toggles created - Features that are deployed but not released have toggles
- [ ] Design QA passed - All visual UI looks and works the way it’s supposed to
- [ ] Usability validated - Team ensures the working code is easy to use
- [ ] Accessibility tests passed - System passes automated and manual accessibility tests and supports applicable devices
- [ ] Code refactored for clarity - Developers can understand the work simply by reviewing the code
- [ ] Dependency Rule followed - More important code doesn’t directly depend on less important code
- [ ] Development debt eliminated - UX and code aligns to the team’s latest understanding of the domain
- [ ] Source code merged - Code has been merged into the main branch
- [ ] 90% unit test coverage achieved - Automated test coverage tools indicate coverage of >= 90%
- [ ] Code reviewed - Code is reviewed by at least two other team members before being merged
- [ ] Code quality checks passed - Code passes all automated quality checks
- [ ] Security scans passed - Code passes expected scans for vulnerabilities and compliance issues
- [ ] Threat model updated - The threat model incorporates new threats and remediations
- [ ] Build process updated - Automated build process include code supporting the user story
- [ ] Load/performance tests passed - Performance tests include the functionality of the user story
- [ ] Documents are updated - All required documentation is up to date and version controlled
- [ ] Code is deployed - Code is deployed to the highest production like environment

### CODE REVIEW

File: backend/functions/lib/testing/mock-gateways/mock-oauth2-gateway.ts [hunk 1]
```diff
@@ -57,6 +57,7 @@ export async function verifyToken(accessToken: string): Promise<CamsJwt> {
 function addSuperUserOffices(user: CamsUser) {
```
Comment (btposey) [inline backend/functions/lib/testing/mock-gateways/mock-oauth2-gateway.ts:57]: Better name?

File: common/src/cams/history.ts [hunk 2]
```diff
@@ -20,13 +20,11 @@ export function isConsolidationHistory(history: unknown): history is Consolidati
 }
 
 // TODO: Consider a way to make the occurredAtTimestamp optional when creating a record, otherwise it is required.
```
Comment (btposey) [inline common/src/cams/history.ts:22]: Remove. Superseded by the `Auditable` type.

File: backend/functions/lib/use-cases/orders/orders.test.ts [hunk 3]
```diff
@@ -68,19 +76,33 @@ describe('Orders use case', () => {
...
+    const mockTransferOrders = [mockTransfer1, mockTransfer2];
+    const mockConsolidation1 = MockData.getConsolidationOrder({
+      override: { orderDate: '2024-08-02' },
+    });
+    const mockConsolidation2 = MockData.getConsolidationOrder({
+      override: { orderDate: '2024-09-02' },
     });
+    const mockConsolidationOrders = [mockConsolidation1, mockConsolidation2];
+
+    // TODO: We need to change getOrders to search similar to how we changed getAll to search
```
Comment (fmaddenflx) [inline backend/functions/lib/use-cases/orders/orders.test.ts:94]: stale.  remove

**Review by fmaddenflx — APPROVED** [review]

File: user-interface/src/lib/models/api.test.ts [hunk 4]
```diff
@@ -4,6 +4,9 @@ import * as httpAdapter from '../utils/http.adapter';
 import { vi } from 'vitest';
 
 describe('Specific tests for the API model', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
```
Comment (jamesobrooks) [inline user-interface/src/lib/models/api.test.ts:9]:
```suggestion
  });
  
```

File: user-interface/src/data-verification/transfer/PendingTransferOrder.test.tsx [hunk 5]
```diff
@@ -233,7 +233,7 @@ describe('PendingTransferOrder component', () => {
       const patchSpy = vi.spyOn(Api2, 'patchTransferOrder').mockResolvedValue({
         data: MockData.getTransferOrder({
           override: {
-            dateFiled: order.dateFiled,
+            dateFiled: '2024-08-01',
```
Comment (amorrow-flexion) [inline user-interface/src/data-verification/transfer/PendingTransferOrder.test.tsx:236]: affix this with the date filed in the below test.

File: backend/functions/lib/testing/mock-gateways/mock-oauth2-gateway.ts (hunk 1 above)
Comment (amorrow-flexion) [reply backend/functions/lib/testing/mock-gateways/mock-oauth2-gateway.ts:57]: `addOfficesToSuperUser` or `assembleSuperUser`?

**Pull request merged**